components/omega/src/ocn/Eos.cpp

```diff
@@ -0,0 +1,253 @@
+   Config *OmegaConfig = Config::getOmegaConfig();
+   Config EosConfig("Eos");
+   Err = OmegaConfig->get(EosConfig);
+   if (Err != 0) {
+      LOG_CRITICAL("Eos: Eos group not found in Config");
+      return Err;
+   }
+   std::string EosTypeStr;
+   Err = EosConfig.get("EosType", EosTypeStr);
+   if (Err != 0) {
+      LOG_CRITICAL("Eos: EosType not found in "
+                   "EosConfig");
+      return Err;
+   }
+
+   if (EosTypeStr == "Linear" or EosTypeStr == "linear") {
+      DefaultEos->EosChoice = EosType::Linear;
+      Err                   = EosConfig.get("LinearDRhoDT",
+                                            DefaultEos->computeSpecVolLinear.dRhodT);
+      if (Err != 0) {
+         LOG_CRITICAL("Eos: linear dRhodT not found in "
+                      "EosConfig");
+         return Err;
+      }
+      Err = EosConfig.get("LinearDRhoDS",
+                          DefaultEos->computeSpecVolLinear.dRhodS);
+      if (Err != 0) {
+         LOG_CRITICAL("Eos: linear dRhodS not found in "
+                      "EosConfig");
+         return Err;
+      }
+      Err = EosConfig.get("LinearRhoT0S0",
+                          DefaultEos->computeSpecVolLinear.RhoT0S0);
+      if (Err != 0) {
+         LOG_CRITICAL("Eos: Ref Rho linearRhoT0S0 not found in "
+                      "EosConfig");
+      }
+   } else if ((EosTypeStr == "teos10") or (EosTypeStr == "teos-10") or
+              (EosTypeStr == "TEOS-10")) {
+      DefaultEos->EosChoice = EosType::TEOS10Poly75t;
+   } else {
+      LOG_CRITICAL("Eos: Unknown EosType requested");
+      Err = -1;
+      return Err;
+   }
+
+   return Err;
+} // end init
+
+void Eos::computeSpecVol(const Array2DReal &SpecVol,
+                         const Array2DReal &ConservativeTemperature,
+                         const Array2DReal &AbsoluteSalinity,
+                         const Array2DReal &Pressure) const {
+   OMEGA_SCOPE(LocSpecVol, SpecVol);
+   OMEGA_SCOPE(LocComputeSpecVolLinear, computeSpecVolLinear);
+   OMEGA_SCOPE(LocComputeSpecVolTEOS10Poly75t, computeSpecVolTEOS10Poly75t);
+   deepCopy(LocSpecVol, 0.0);
+   if (EosChoice == EosType::Linear) {
+      parallelFor(
+          "eos-linear", {NCellsAll, NChunks},
+          KOKKOS_LAMBDA(I4 ICell, I4 KChunk) {
+             LocComputeSpecVolLinear(LocSpecVol, ICell, KChunk,
+                                     ConservativeTemperature, AbsoluteSalinity);
+          });
+   } else if (EosChoice == EosType::TEOS10Poly75t) {
+      parallelFor(
+          "eos-teos10", {NCellsAll, NChunks},
+          KOKKOS_LAMBDA(I4 ICell, I4 KChunk) {
+             LocComputeSpecVolTEOS10Poly75t(LocSpecVol, ICell, KChunk,
+                                            ConservativeTemperature,
+                                            AbsoluteSalinity, Pressure);
+          });
+   }
+}
+
+//------------------------------------------------------------------------------
+// Define IO fields and metadata
+void Eos::defineFields() {
```

Comment:
Do we actually want the EOS to own SpecVol variables? Or should they sit somewhere else?  If here, then we may need to add others (eg a surface-displaced variable). And compute functions to match each.

Reply:
I think it should live here.  I'm not sure we need a surace displaced value.  KPP requires a surface layer average displaced downward - so a generic interface that displaces by 'k' layers